### DISCUSSION TITLE
are there any recorder tool to generate locator and code for pywinauto?

### DISCUSSION THREAD

**Discussion by yongjuntian** [pywinauto/pywinauto#1233] [Ideas]
are there any recorder tool to generate locator and code for pywinauto? if there are some recorder tools, it should improve the development efficiancy

**Comment by beuaaa** [comment]
There has been a development attempt in the official Github repository  (see issue https://github.com/pywinauto/pywinauto/issues/332) but it is a difficult development and it seems to be stalled.

I myself have developed a tool called "Pywinauto Recorder". It is an independent project from Pywinauto. There is still a lot of work to do, but it is progressing little by little. Here is the [link](https://github.com/beuaaa/pywinauto_recorder) of the Github repository. 

In the [documentation page ](https://pywinauto-recorder.readthedocs.io/ )there is a [demonstration video](https://www.youtube.com/watch?v=GQiT5w4dzgE).

Pywinauto Recorder does not directly produce Pywinauto code but a code that is an overlay of Pywinauto. 
All the functions of this generated code return a Pywinauto wrapper so it is possible to enhance the generated code with Pywinauto code.

Here is an example of code produced by "Pywinauto Recorder":
``` Python
wrapper = find(u"Calculator||Window->Calculator||Window->||Custom->||Group->Number pad||Group->Zero||Button")
wrapper.draw_outline()
```
This code finds the Pywinauto wrapper of the Zero button of the Windows 11 calculator (thanks to the full UI path) and calls the Pywinauto draw_outline function.

I wish I had an equivalent Pywinauto code like this:
``` Python
from pywinauto.application import Application
app = Application(backend="uia")
app.connect(title="Calculator")
wrapper = app['CalculatorWindow']['CalculatorWindow']['Custom']['Group']['Number padGroup']['ZeroButton']
wrapper.draw_outline()
```
But the Pywinauto code doesn't work like that. I would like to have proposals for equivalent Pywinauto code from active members of the Pywinauto community.
Pywinauto Recorder is intended to generate Pywinauto code directly, any help to further develop this tool is welcome.
